### PR TITLE
Added Recent history in cart page

### PR DESCRIPTION
**Pull request by akathuria289**
## Issue Reference
#6373

## Description
Added Recent History of the customer in the customer cart page

### CODE REVIEW

**Comment by Sanjaybhattwebkul** [comment]
@akathuria289 
**Not working**, in mobile view the Recent history is **not visible**.

**Comment by akathuria289** [comment]
@Sanjaybhattwebkul Check Now

**Comment by Sanjaybhattwebkul** [comment]
@akathuria289 
The cart summary should not visible after recent history. please find the screenshot [here](https://prnt.sc/gJeBdGPYnUq3)

**Comment by akathuria289** [comment]
@Sanjaybhattwebkul Not getting any issues Please re-check 
Shopping-Cart (3)
Shopping-Cart (2)

**Comment by Sanjaybhattwebkul** [comment]
@akathuria289 
Getting **exceptions**, while trying to visit the **shopping cart**. Please find the screenshot [Here](https://webkul.chatwhizz.com/share/screenshot/63fc3b46bf991905516ced01)

**Comment by Sanjaybhattwebkul** [comment]
@akathuria289 
Getting **exception**, while trying to run a **project** with the **IP address**. Please find the screenshot [Here](https://prnt.sc/VHqy80xyRRuz)

**Comment by akathuria289** [comment]
@Sanjaybhattwebkul Done

**Comment by Sanjaybhattwebkul** [comment]
@akathuria289 
Getting **exception**, while guest user trying to **visit** the **shopping cart.** please find the **screenshot** [Here](https://prnt.sc/M7zszABengDD)

**Comment by akathuria289** [comment]
@Sanjaybhattwebkul Please check NOW

**Comment by Sanjaybhattwebkul** [comment]
@akathuria289 
**Still not working**, getting the same issue

**Comment by Sanjaybhattwebkul** [comment]
@akathuria289 
**Getting the exception** while the customer is trying to **visit** the shopping **cart** and the customer has **no order yet**. Please find the **screenshot** [Here](https://prnt.sc/lU01uFNhLO05)

**Comment by Sanjaybhattwebkul** [comment]
@akathuria289 
**Now working** fine, Please find the **screenshot** [Here](https://webkul.chatwhizz.com/share/screenshot/6401ac626645cb5bafb40a81)

**Review by Sanjaybhattwebkul — DISMISSED** [review]

File: packages/Webkul/Sales/src/Repositories/OrderItemRepository.php [hunk 1]
```diff
@@ -226,4 +227,35 @@ public function updateProductOrderedInventories($orderItem)
...
+    /**
+     * Get customer order history
+     *
+     * @param array $orderIds
+     * @return Webkul\Sales\Repositories\OrderItemRepository
+     */
+    public function getCustomerHistory($orderIds) 
+    {
+        return $this->model->whereIn('order_id', $orderIds)
+                            ->distinct()
```
Comment (jitendra-webkul) [inline packages/Webkul/Sales/src/Repositories/OrderItemRepository.php:240]: Fixed indentation

Comment (akathuria289) [reply packages/Webkul/Sales/src/Repositories/OrderItemRepository.php]: Done

File: packages/Webkul/Sales/src/Repositories/OrderItemRepository.php [hunk 2]
```diff
@@ -226,4 +226,35 @@ public function updateProductOrderedInventories($orderItem)
...
+     * @param array $orderIds
+     * @return Webkul\Sales\Repositories\OrderItemRepository
+     */
+    public function getOrderItems($orderIds) 
+    {
+        return $this->model->whereIn('order_id', $orderIds)
+               ->distinct()
+               ->groupBy('product_id')
+               ->get();
+    }
```
Comment (devansh-webkul) [inline packages/Webkul/Sales/src/Repositories/OrderItemRepository.php:242]: This needs to be fixed as we discussed.

File: packages/Webkul/Checkout/src/Models/CartItem.php [hunk 3]
```diff
@@ -29,11 +29,17 @@ class CartItem extends Model implements CartItemContract
...
+     * Get Product
+     */
     public function product(): HasOne
     {
         return $this->hasOne(ProductProxy::modelClass(), 'id', 'product_id');
     }
 
+    /**
+     * Get Cart
+     */
```
Comment (devansh-webkul) [inline packages/Webkul/Checkout/src/Models/CartItem.php:42]: And return doc to both methods.

File: packages/Webkul/Shop/src/Http/Controllers/ProductsCategoriesProxyController.php [hunk 4]
```diff
@@ -49,6 +50,19 @@ public function index(Request $request)
...
+
+                    if (! isset($productsData)) {
+                        Cookie::queue(Cookie::make('product', $productArr, time() + (86400 * 30), "/"));
+                        
+                    } else {
+                        $productArr = json_decode($productsData);
+                        array_push($productArr, $product->sku);
+                        $productArr = json_encode($productArr);
+                        Cookie::queue(Cookie::make('product', $productArr, time() + (86400 * 30), "/"));
+                    }
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Controllers/ProductsCategoriesProxyController.php:64]: This needs to be improvised as variable namings are not good.

Comment (akathuria289) [reply packages/Webkul/Shop/src/Http/Controllers/ProductsCategoriesProxyController.php]: fixed

File: packages/Webkul/Checkout/src/Models/CartItem.php (hunk 3 above)
Comment (akathuria289) [reply packages/Webkul/Checkout/src/Models/CartItem.php]: fixed

**Comment by Amitk-webkul** [comment]
@akathuria289,
Please resolve the conflicts.

**Comment by akathuria289** [comment]
@Amitk-webkul Fixed

**Comment by Amitk-webkul** [comment]
1. If the layout has more content so both layouts should scroll the same as a single page.


layout

2. It is showing an Apostrophe sign below apply coupon button.

sign

**Comment by Sanjaybhattwebkul** [comment]
@akathuria289 
Getting the **exception**. while trying to visit the **shopping cart**. please find the screenshot [Here](https://webkul.chatwhizz.com/share/screenshot/6421606f2a666707c4f7c976)

**Comment by Amitk-webkul** [comment]
@akathuria289 Please resolve the conflicts.

**Comment by Amitk-webkul** [comment]
Issues 3 and 4 are fixed but getting exceptions while trying to visit the shopping cart is still there.

Also, It is not showing compare list in the shopping cart. [Here](https://tinyurl.com/2zkyzmjw)

Also, While new user is register it is not showing recently views tab in the shopping cart.

**Comment by Amitk-webkul** [comment]
The exception issue is fixed.

When a new user registers and go to the product page and add to cart the product after that while visiting the shopping cart. here it is not showing a recent view.[Here](https://tinyurl.com/2ea8qbvu)

**Review by Sanjaybhattwebkul — APPROVED** [review]

**Comment by devansh-webkul** [comment]
This will not be done at this stage.

**Pull request closed without merging**